block-storage-attacher/tests/e2e/e2e-tests/runon_worker.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+#. ./connect_worker.sh  worker private ip bash
+if [ $# -lt 2 ]; then
+  echo "runon [node IP] [command]"
+  exit 1;
+fi
+
+function cleanup {
+  rm -rf ${TEMP_DIR}
+}
+
+function pod_status {
+  kubectl get pods $1 -n ${NAMESPACE} -o json | jq '.status.conditions[] | select(.type == "Ready") | .status ' | sed 's/\"//g'
+}
+
+# create a random pod name
+JOB_NAME=$(LC_CTYPE=C cat /dev/urandom | base64 | tr -dc a-z0-9 | fold -w 32 | head -n 1)
+NAMESPACE="ibm-system"
+
+# create a tempdir
+TEMP_DIR=$(mktemp -d)
+trap cleanup EXIT
+
+NODE=$1
+
+shift
+if [ "$1" == "--" ]; then
+  shift
+fi
+
+COMMAND=$@
+
+(cat << EOF
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: ${JOB_NAME}
+  namespace: ${NAMESPACE}
+  labels:
+    app: runon-shell
+spec:
+  template:
+    spec:
+      nodeSelector:
+        kubernetes.io/hostname: ${NODE}
+      containers:
+        - name: runon
+          image: "debian:jessie"
+          command:
+            - sleep
+            - 8h
+          securityContext:
+            privileged: true
+      hostPID: true
+      restartPolicy: Never
+EOF
+) > ${TEMP_DIR}/job.yml
+
+if ! kubectl apply -f ${TEMP_DIR}/job.yml 2>&1 > /dev/null; then
+  echo "unable to create job, bailing out"
+  exit 1
+fi
+
+
+# get the uid
+ID=$(kubectl get job ${JOB_NAME} -n ${NAMESPACE} -o 'jsonpath={.metadata.uid}')
+if [ -z "${ID}" ]; then
+  echo "ERR unable to get job id"
+  exit 1
+fi
+
+
+POD=$(kubectl get pods -n ${NAMESPACE} -l controller-uid=${ID},job-name=${JOB_NAME} -o 'jsonpath={.items[].metadata.name}')
+
+SUCCESS=
+current_time=$(date +%s)
+stop_time=$((current_time + 60)) # this shouldn't take that long, give it 60 seconds
+
+status=$(pod_status ${POD})
+while [[ $current_time -lt $stop_time ]]; do
+  if [ "${status}" = "True" ]; then
+    SUCCESS=true
+    break
+  else
+    sleep 1
+  fi
+  status=$(pod_status ${POD})
+done
+
+if [ "${SUCCESS}" = "true" ]; then
+  kubectl exec -it "${POD}" -n ${NAMESPACE} -- nsenter -t 1 -m -u -i -n -p -- $@ <&0 2>&1
```

Comment:
After the `kubectl exec` is done, the JOB created earlier using`$JOB_NAME` needs to be deleted. Pasting a sample command for reference.
`kubectl -n kube-system delete job 2rpg7tzot76tup747ze7roskroh7xwcz`. 
`kubectl -n kube-system delete job $JOB_NAME`.  Login to cluster and confirm if the job is deleted on cluster just for initial testing.

Reply:
ok i will check and update the job is being deleted . If we take the bash terminal then job exists else once the command completes it is killed I think . I will check

Reply:
@mumutyal : Once you create a job using apply kubectl command, the job gets created. Its not created upon exec. So it will definitely be present and hence after exec add a delete job cmd to delete it. Test this properly if e2e is deleting jobs by manually using kubectl cmds.